mongodbatlas/fw_resource_mongodbatlas_encryption_at_rest.go

```diff
@@ -0,0 +1,566 @@
+						},
+						"subscription_id": schema.StringAttribute{
+							Optional:  true,
+							Sensitive: true,
+						},
+						"resource_group_name": schema.StringAttribute{
+							Optional: true,
+						},
+						"key_vault_name": schema.StringAttribute{
+							Optional: true,
+						},
+						"key_identifier": schema.StringAttribute{
+							Optional:  true,
+							Sensitive: true,
+						},
+						"secret": schema.StringAttribute{
+							Optional:  true,
+							Sensitive: true,
+						},
+						"tenant_id": schema.StringAttribute{
+							Optional:  true,
+							Sensitive: true,
+						},
+					},
+				},
+			},
+			"google_cloud_kms_config": schema.ListNestedBlock{
+				Validators: []validator.List{listvalidator.SizeAtMost(1)},
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"enabled": schema.BoolAttribute{
+							Optional: true,
+							Computed: true,
+							PlanModifiers: []planmodifier.Bool{
+								boolplanmodifier.UseStateForUnknown(),
+							},
+						},
+						"service_account_key": schema.StringAttribute{
+							Optional:  true,
+							Sensitive: true,
+						},
+						"key_version_resource_id": schema.StringAttribute{
+							Optional:  true,
+							Sensitive: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *EncryptionAtRestRS) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var encryptionAtRestPlan *tfEncryptionAtRestRSModel
+	var encryptionAtRestConfig *tfEncryptionAtRestRSModel
+	conn := r.client.Atlas
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &encryptionAtRestPlan)...)
+	resp.Diagnostics.Append(req.Config.Get(ctx, &encryptionAtRestConfig)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectID := encryptionAtRestPlan.ProjectID.ValueString()
+	encryptionAtRestReq := &matlas.EncryptionAtRest{
+		GroupID: projectID,
+	}
+	if encryptionAtRestPlan.AwsKmsConfig != nil {
+		encryptionAtRestReq.AwsKms = *newAtlasAwsKms(encryptionAtRestPlan.AwsKmsConfig)
+	}
+	if encryptionAtRestPlan.AzureKeyVaultConfig != nil {
+		encryptionAtRestReq.AzureKeyVault = *newAtlasAzureKeyVault(encryptionAtRestPlan.AzureKeyVaultConfig)
+	}
+	if encryptionAtRestPlan.GoogleCloudKmsConfig != nil {
+		encryptionAtRestReq.GoogleCloudKms = *newAtlasGcpKms(encryptionAtRestPlan.GoogleCloudKmsConfig)
+	}
+
+	var encryptionResp *matlas.EncryptionAtRest
+	var err error
+	for i := 0; i < 5; i++ {
+		encryptionResp, _, err = conn.EncryptionsAtRest.Create(ctx, encryptionAtRestReq)
+		if err != nil {
+			if strings.Contains(err.Error(), "CANNOT_ASSUME_ROLE") || strings.Contains(err.Error(), "INVALID_AWS_CREDENTIALS") ||
+				strings.Contains(err.Error(), "CLOUD_PROVIDER_ACCESS_ROLE_NOT_AUTHORIZED") {
+				log.Printf("warning issue performing authorize EncryptionsAtRest not done try again: %s \n", err.Error())
+				log.Println("retrying ")
+				time.Sleep(10 * time.Second)
+				encryptionAtRestReq.GroupID = projectID
+				continue
```

Comment:
Was this logic already here? Can we use `retry.StateChangeConf` instead? you can file a ticket to improve the logic to use `retry.StateChangeConf`

Reply:
yes, this has been migrated as is. Created https://jira.mongodb.org/browse/INTMDB-1039